src/centroiders.cpp

```diff
@@ -92,6 +94,84 @@ int BasicThreshold(unsigned char *image, int imageWidth, int imageHeight) {
     return mean + (std * 5);
 }
 
+int Limit(bool test, int i, int leftover, int div) {
+    if(test) {
+        return i * (div + 1);
+    } else {
+        return leftover * (div + 1) + (i - leftover) * div;
+    }
+}
+// pixel * divisions/ size floor
+// subdivisionsize * size / divisions ceil
+
+// Box is which subdivision
+int Box(int box, unsigned char *image, int imageWidth, int imageHeight, int subdivisions, int horizontalLeftover, int horizontalDiv, int verticalLeftover, int verticalDiv) {
+    int row = box / subdivisions; // Finds which row on the subdivisions we're in
+    int col = box % subdivisions; // Finds the column on the subdivisions we're in
+    double average = 0;
+    double squareSum = 0;
+    long count = 0;
+    //std::cout << "Box " << box << "\n";
+    //std::cout << Limit(row < horizontalLeftover, row, horizontalLeftover, horizontalDiv) << " " << Limit(row < horizontalLeftover, row + 1, horizontalLeftover, horizontalDiv) << "\n";
+    //std::cout << Limit(col < verticalLeftover, col, verticalLeftover, verticalDiv) << " " << Limit(col < verticalLeftover, col + 1, verticalLeftover, verticalDiv) << "\n";
+    //std::cout << "\n";
+    int secondCount = 0;
+    int number;
+    for(int i = Limit(row < horizontalLeftover, row, horizontalLeftover, horizontalDiv); i < Limit(row < horizontalLeftover, row + 1, horizontalLeftover, horizontalDiv); i++) {
+        for(int j = Limit(col < verticalLeftover, col, verticalLeftover, verticalDiv); j < Limit(col < verticalLeftover, col + 1, verticalLeftover, verticalDiv); j++) {
+            average += image[i * imageWidth + j];
+            squareSum += image[i * imageWidth + j] * image[i * imageWidth + j];
+            count++;
+
+            // Checks for Error
+            if(FindSubdivision(i * imageWidth + j, imageWidth, imageHeight, subdivisions) != box) {
+                secondCount++;
+                number = FindSubdivision(i * imageWidth + j, imageWidth, imageHeight, subdivisions);
+            }
```

Comment:
It's okay to leave some error checking code in the final product, but it should be simplified to use `assert`, for example:

`assert(FindSubdivision(i*imageWidth+j, imageWidth, imageHeight, subdivisions) == box)`

This has two advantages:
+ We can disable assertions using an extra command-line flag to `make`, so that it doesn't affect the speed of our code.
+ It's obvious to anyone reading the code that it's only meant for error checking.